osc-server/src/main/java/org/osc/core/broker/service/securitygroup/BindSecurityGroupService.java

```diff
@@ -324,4 +343,107 @@ private VirtualSystemPolicyBindingDto findDuplicateServiceByOrder(List<VirtualSy
+    		map.put("Id", request.getSfcId());
+    		if(ValidateUtil.isFieldNull(map)) {
+    			//user wants to unbind serviceFunctonChain with Security Group
+    			return null;
+    		}
+    		
+    		VirtualizationConnector vc = VirtualizationConnectorEntityMgr.findById(em, request.getVcId());
+    		sfc = em.find(ServiceFunctionChain.class, request.getSfcId());
+    		if (sfc == null) {
+
+    			throw new VmidcBrokerValidationException(
+    					"Service Function Chain entry with Id: " + request.getSfcId() + " is not found.");
+    		}
+    		
+    		 ValidateUtil.checkMarkedForDeletion(this.securityGroup, this.securityGroup.getName());
+    		 
+			    		
+    		//sfc parent and vcid check
+    		if(!sfc.getVirtualizationConnector().getId().equals(vc.getId())){
+    			throw new VmidcBrokerValidationException(
+    					"Service Function Chain Virutalization Connector Id " + sfc.getVirtualizationConnector().getId() + 
+    								" do not match with  given Virutalization Connector Id " 
+    								+ vc.getId() + " ");
+    		}
+    		
+    		//sfc chain should not be empty
+    		List<Long> sfcVsIdList = new ArrayList<Long>();
+    		for(VirtualSystem vs : CollectionUtils.emptyIfNull(sfc.getVirtualSystems())) {
+    			sfcVsIdList.add(vs.getId());
+    		}
+			List<VirtualSystemPolicyBindingDto> servicesToBindTo = request.getServicesToBindTo();
+
+             // For all services selected, create or update the security group interfaces
+             for (VirtualSystemPolicyBindingDto serviceToBindTo : servicesToBindTo) {
+                 Long virtualSystemId = serviceToBindTo.getVirtualSystemId();
+                 if(null == sfcVsIdList || 0 == sfcVsIdList.size() || !sfcVsIdList.contains(virtualSystemId)) {
+                	 throw new VmidcBrokerValidationException(
+         					"Binding request Virutal Systm Id : " + virtualSystemId + 
+         					" do not match with any of the id in Service Function Chain's Virtual System Id list"); 
+                 }
+                 //if this Virtual system is pointing to more than one sfc , we need to throw exception while binding.
+                 //each service function chain VNF(VS) can only be part of one service chain at a given time.
+                 //Do we need to modify sfc<-->VS map from many to many to one to many (??).
+                 VirtualSystem vs = VirtualSystemEntityMgr.findById(em, virtualSystemId);
+                 //if this virtual system is pointing to more than one SFC throw exception
+                 if(vs.getServiceFunctionChains().size() > 1) {
+                	 throw new VmidcBrokerValidationException(
+          					"Virtual system " + virtualSystemId + " should not be part of more than one chain");  
+                 }
+             }
+            
+             //number of vsIds in sfc should match with number of binding request vsIds
+             if(sfcVsIdList.size() != servicesToBindTo.size()) {
```

Comment:
remove size check. Remove an entry from sfcidlist and make sure the sfcidList is empty at the end